File: Examples/Java/SDK/src/main/java/com/aspose/cells/cloud/examples/text/GetTextItemWorkbook.java
package com.aspose.cells.cloud.examples.text;

import com.aspose.cells.api.CellsApi;
import com.aspose.cells.cloud.examples.Configuration;
import com.aspose.cells.cloud.examples.Utils;
import com.aspose.storage.api.StorageApi;

import java.io.IOException;
import java.nio.file.Path;

public class GetTextItemWorkbook {

    public static void main(String... args) throws IOException {
        // ExStart: get-text-item-workbook

        try {
            // Instantiate Aspose Storage API SDK
            StorageApi storageApi = new StorageApi(Configuration.apiKey, Configuration.appSID, true);

            // Instantiate Aspose Words API SDK
            CellsApi cellsApi = new CellsApi(Configuration.apiKey, Configuration.appSID, true);
            String input = "sample1.xlsx";
            Path inputFile = Utils.getPath(GetTextItemWorkbook.class, input);

            storageApi.PutCreate(input, null, null, inputFile.toFile());

            com.aspose.cells.model.TextItemsResponse apiResponse = cellsApi.GetWorkBookTextItems(input, null, null);

            for (com.aspose.cells.model.TextItem item : apiResponse.getTextItems().getTextItemList()) {
                System.out.println("Text: " + item.getText());
                System.out.println("Href: " + item.getLink().getHref());
                System.out.println("");
            }

        }

        catch (Exception e) {
            e.printStackTrace();
        }
        // ExEnd: get-text-item-workbook
    }

}

File: Examples/DotNet/CSharp/Text/GetTextItemWorkbook.cs
using System;
using Com.Aspose.Cells.Api;
using Com.Aspose.Cells.Model;
using Com.Aspose.Storage.Api;

namespace Text
{
    class GetTextItemWorkbook
    {
        public static void Run()
        {
            // ExStart:1
            CellsApi cellsApi = new CellsApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);
            StorageApi storageApi = new StorageApi(Common.APP_KEY, Common.APP_SID, Common.BASEPATH);

            String fileName = "Sample_Test_Book.xls";
            String storage = null;
            String folder = null;

            try
            {
                // Upload source file to aspose cloud storage
                storageApi.PutCreate(fileName, "", "", System.IO.File.ReadAllBytes(Common.GetDataDir() + fileName));

                // Invoke Aspose.Cells Cloud SDK API to get text items from workbook
                TextItemsResponse apiResponse = cellsApi.GetWorkBookTextItems(fileName, storage, folder);

                if (apiResponse != null && apiResponse.Status.Equals("OK"))
                {
                    foreach (TextItem item in apiResponse.TextItems.TextItemList)
                    {
                        Console.WriteLine("Text: " + item.Text);
                        Console.WriteLine("Href: " + item.link.Href);
                    }
                    Console.ReadKey();
                }
            }
            catch (Exception ex)
            {
                System.Diagnostics.Debug.WriteLine("error:" + ex.Message + "\n" + ex.StackTrace);
            }
            // ExEnd:1
        }
    }
}
